### PR TITLE
[Docs] Update TypeChecker.rst with a list of not-yet-ported diagnostics

### PR DESCRIPTION
**Pull request by xedin**

### CODE REVIEW

File: docs/TypeChecker.rst [hunk 1]
```diff
@@ -971,3 +971,34 @@ well-typed suggestions.
...
+---------------------------
+
+We are currently working on porting type-check based diagnostics over
+to the new diagnostic framework, which is described in detail in this
+`blog post
+<https://swift.org/blog/new-diagnostic-arch-overview/>`_.
+
+The things in the queue yet to be ported are:
+
+- ``visitTupleExpr``: Diagnostics related to label/type mismatches
```
Comment (hborla) [inline docs/TypeChecker.rst]: Should we remove this since I'm working on it now?

Comment (xedin) [reply docs/TypeChecker.rst]: I think it would make sense to remove once the work is complete?

**Comment by xedin** [comment]
@swift-ci please smoke test

**Review by hborla — APPROVED** [review]

**Review by DougGregor — APPROVED** [review]

**Comment by slavapestov** [comment]
That's a short list. Excellent work @hborla and @xedin.

File: docs/TypeChecker.rst [hunk 2]
```diff
@@ -971,3 +971,34 @@ well-typed suggestions.
...
+  located in ``ConstraintSystem::simplifyMemberConstraint``.
+
+- Closure expression diagnostics: ``diagnoseClosureExpr``.
+
+- Diagnostics related to ``if`` statement - "conditional" type mismatch
+  and, in case of ternary operator, type mismatches between branches.
+
+- Problems related to calls and operator applications e.g.
+
+  - ``@dynamicallyCallable`` related diagnostics
```
Comment (dan-zheng) [inline docs/TypeChecker.rst]:
```suggestion
  - ``@dynamicCallable`` related diagnostics
```

Comment (xedin) [reply docs/TypeChecker.rst]: Thank you!

**Comment by xedin** [comment]
@swift-ci please smoke test

**Pull request merged**